packages/api/src/providers/fitbit.ts

```diff
@@ -363,63 +370,50 @@ export class Fitbit extends Provider implements OAuth2 {
 
   async postAuth(token: string, user: ConnectedUser): Promise<void> {
     const { access_token: accessToken, user_id: fitbitUserId, scope: scopes } = JSON.parse(token);
-
-    const subscriptionTypes: Record<FitbitScopes, FitbitCollectionTypesWithoutUserRevokedAccess> = {
-      [FitbitScopes.activity]: "activities",
-      [FitbitScopes.nutrition]: "foods",
-      [FitbitScopes.sleep]: "sleep",
-      [FitbitScopes.weight]: "body",
-    };
-
-    const activeSubscriptions = await this.getActiveSubscriptions(accessToken);
-    if (activeSubscriptions)
-      await this.revokeTokenFromOtherUsers(user, fitbitUserId, activeSubscriptions, accessToken);
-
-    // TODO #652: Implement userRevokedAccess
+    await this.revokeTokenFromOtherUsers(user, fitbitUserId);
 
     // Creates new WH subscriptions based on the user's selected scopes
     await Promise.all(
-      Object.entries(subscriptionTypes).map(async ([key, subscriptionType]) => {
+      Object.entries(Fitbit.subscriptionTypes).map(async ([key, subscriptionType]) => {
         if (scopes.includes(key)) {
           const subscriptionId = `${fitbitUserId}-${subscriptionType}`;
           const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}.json`;
-          this.createSubscription(subscriptionUrl, accessToken);
+          await this.createSubscription(subscriptionUrl, accessToken);
         }
       })
     );
```

Comment:
I know it was already like this, but shouldn't we use `Promise.allSettled` so we have a more deterministic behavior?

`Promise.all` returns upon the first error, so we don't know whether the other ones executed or not.

With `Promise.allSettled` we know exactly which ones worked and which ones failed.

Reply:
Ah, I missed that part! Good catch!